src/rgw/rgw_op.cc

```diff
@@ -110,23 +112,47 @@ static int parse_range(const char *range, off_t& ofs, off_t& end, bool *partial_
     goto done;
 
   *partial_content = true;
+  end = -2;
 
+  if (pos == 0 && s.length() == 1)
+    goto done;
   ofs_str = s.substr(0, pos);
   end_str = s.substr(pos + 1);
+  for (il = ofs_str.length(); il > 0; --il)
+    if (!isspace(ofs_str[il-1])) break;
```

Comment:
the code above this already uses isspace().  Also strtoll internally uses isspace().  And with "find_last_not_of" there's an icky problem with deciding exactly which "whitespace" characters one wants to count and I don't think we want that to be different from what strtoll & the isspace code above does.